### ISSUE TITLE
Provide a Docker Hub Mirror for use with the GitHub Action runners

### ISSUE THREAD

**Issue by TimWolla** [ubicloud/ubicloud#2245]
Today my GitHub Action Workflows started failing in the “Initialize Containers” stage, because apparently Ubicloud’s runners started to get rate limited by Docker Hub:

```
Starting redis service container
  /usr/bin/docker pull redis
  Using default tag: latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 2.085 seconds before retry.
  /usr/bin/docker pull redis
  Using default tag: latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 1.829 seconds before retry.
  /usr/bin/docker pull redis
  Using default tag: latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Error: Docker pull failed with exit code 1
```

Unfortunately it's not possible to reconfigure Docker Daemon before the service containers are set up, so I was unable to configure the `registry-mirrors` setting to provide a transparent pull through cache as explained in https://docs.docker.com/docker-hub/mirror/#configure-the-docker-daemon.

I was able to work around the issue by manually changing the `image:` reference to the full name on a pull through cache, but that required changes in multiple locations and also requires to remember to never use the regular image name anywhere.

-----------------

I expect that other Ubicloud users will also run into this issue sooner or later and thus would like to suggest that Ubicloud provides a pull-through cache for the public images on Docker Hub and configures it by default in the provided images.

**Comment by fdr** [comment]
Yeah, we've noticed a sudden change here in dockerhub's behavior and got several reports, we're trying to do a workaround

**Comment by fdr** [comment]
FYI, the workaround we've settled on for now is here, we're not sure precisely why the quota has bitten us so hard so suddenly. https://github.com/ubicloud/ubicloud/discussions/2244#discussioncomment-11231599

We'll have to figure out a mirroring situation, as you write.

**Comment by enescakir** [comment]
We resolved the root cause of DockerHub rate limit issue.

Root cause: https://github.com/ubicloud/ubicloud/discussions/2244#discussioncomment-11232417
